tests/resource/aws_ssm_resource_data_sync: Implement sweeper

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

The acceptance testing can leave lingering resources on failed runs (e.g. via coding issues during development or panics). These lingering resources can cause quota limits to be reached, extraneous money to be spent, or in some worst cases leave open security risks in the testing account.

Currently testing is failing in the main testing account due to hitting the service limit:

```
=== CONT  TestAccAWSSsmResourceDataSync_update
resource_aws_ssm_resource_data_sync_test.go:40: Step 1/3 error: Error running apply: exit status 1
Error: ResourceDataSyncCountExceededException: You've already reached the maximum of 5 Resource Data Sync Configurations.
on terraform_plugin_test.tf line 47, in resource "aws_ssm_resource_data_sync" "test":
47: resource "aws_ssm_resource_data_sync" "test" {
--- FAIL: TestAccAWSSsmResourceDataSync_update (36.42s)
```

### Affected Resource

* aws_ssm_resource_data_sync

### Definition of Done

- [ ] `go test ./aws -v -timeout=10h -sweep-allow-failures -sweep=us-west-2,us-east-1 -sweep-run=aws_ssm_resource_data_sync` completes successfully without error
- [ ] `aws ssm list-resource-data-sync | jq '.ResourceDataSyncItems | length'` shows no remaining resources

### References

* [Extending Terraform sweeper documentation](https://www.terraform.io/docs/extend/testing/acceptance-tests/sweepers.html)

--- a/aws/resource_aws_ssm_resource_data_sync_test.go
+++ b/aws/resource_aws_ssm_resource_data_sync_test.go
@@ -55,10 +55,8 @@ func testSweepSsmResourceDataSyncs(region string) error {
 		errs = multierror.Append(errs, fmt.Errorf("error listing SSM Resource Data Sync for %s: %w", region, err))
 	}
 
-	if len(sweepResources) > 0 {
-		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
-			errs = multierror.Append(errs, fmt.Errorf("error sweeping SSM Resource Data Sync for %s: %w", region, err))
-		}
+	if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error sweeping SSM Resource Data Sync for %s: %w", region, err))
 	}
 
 	if testSweepSkipSweepError(errs.ErrorOrNil()) {

--- a/aws/resource_aws_ssm_resource_data_sync_test.go
+++ b/aws/resource_aws_ssm_resource_data_sync_test.go
@@ -5,11 +5,69 @@ import (
 	"log"
 	"testing"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ssm"
+	multierror "github.com/hashicorp/go-multierror"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
+
+func init() {
+	resource.AddTestSweepers("aws_ssm_resource_data_sync", &resource.Sweeper{
+		Name: "aws_ssm_resource_data_sync",
+		F:    testSweepSsmResourceDataSyncs,
+	})
+}
+
+func testSweepSsmResourceDataSyncs(region string) error {
+	client, err := sharedClientForRegion(region)
+
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+
+	conn := client.(*AWSClient).ssmconn
+	sweepResources := make([]*testSweepResource, 0)
+	var errs *multierror.Error
+
+	input := &ssm.ListResourceDataSyncInput{}
+
+	err = conn.ListResourceDataSyncPages(input, func(page *ssm.ListResourceDataSyncOutput, lastPage bool) bool {
+		if page == nil {
+			return !lastPage
+		}
+
+		for _, resourceDataSync := range page.ResourceDataSyncItems {
+			r := resourceAwsSsmResourceDataSync()
+			d := r.Data(nil)
+
+			d.SetId(aws.StringValue(resourceDataSync.SyncName))
+			d.Set("name", resourceDataSync.SyncName)
+
+			sweepResources = append(sweepResources, NewTestSweepResource(r, d, client))
+		}
+
+		return !lastPage
+	})
+
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error listing SSM Resource Data Sync for %s: %w", region, err))
+	}
+
+	if len(sweepResources) > 0 {
+		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error sweeping SSM Resource Data Sync for %s: %w", region, err))
+		}
+	}
+
+	if testSweepSkipSweepError(errs.ErrorOrNil()) {
+		log.Printf("[WARN] Skipping SSM Resource Data Sync sweep for %s: %s", region, errs)
+		return nil
+	}
+
+	return errs.ErrorOrNil()
+}
 
 func TestAccAWSSsmResourceDataSync_basic(t *testing.T) {
 	resourceName := "aws_ssm_resource_data_sync.test"

